skynet/modules/ttt/summaries/processor.py

```diff
@@ -1,48 +1,73 @@
-import asyncio
-from concurrent.futures import ThreadPoolExecutor
-
 from langchain.chains.summarize import load_summarize_chain
-from langchain.llms import LlamaCpp
-from langchain.prompts import PromptTemplate
+from langchain.prompts import ChatPromptTemplate
 from langchain.text_splitter import RecursiveCharacterTextSplitter
+from langchain_core.documents import Document
+from langchain_openai import ChatOpenAI
 
-from skynet.env import llama_n_batch, llama_n_gpu_layers, llama_path
+from skynet.env import app_uuid, openai_api_base_url
+from skynet.logs import get_logger
 
-from .prompts.action_items import action_items_template
-from .prompts.summary import summary_template
-from .v1.models import Job, JobType
+from .prompts.action_items import action_items_conversation_prompt, action_items_text_prompt
+from .prompts.summary import summary_conversation_prompt, summary_text_prompt
+from .v1.models import DocumentPayload, HintType, JobType
 
-executor = None
 llm = None
+map_reduce_threshold = 12000
+log = get_logger(__name__)
+
+
+hint_type_to_prompt = {
+    JobType.SUMMARY: {
+        HintType.CONVERSATION: summary_conversation_prompt,
+        HintType.TEXT: summary_text_prompt,
+    },
+    JobType.ACTION_ITEMS: {
+        HintType.CONVERSATION: action_items_conversation_prompt,
+        HintType.TEXT: action_items_text_prompt,
+    },
+}
 
 
 def initialize():
-    global executor, llm
-    executor = ThreadPoolExecutor(max_workers=1)
-
-    llm = LlamaCpp(
-        model_path=llama_path,
-        temperature=0.01,
-        max_tokens=4096,
-        n_ctx=4096,
-        n_gpu_layers=llama_n_gpu_layers,
-        n_batch=llama_n_batch,
+    global llm
```

Comment:
Is this initialize necessary? Can't we just create it at imprt time?

Reply:
if we move it like that, we need to also better isolate the summaries:dispatcher module to make sure it's not importing it indirectly.

Reply:
Gotcha. No strong opinion then.